src/FsAutoComplete.Core/FsprojEdit.fs

```diff
@@ -0,0 +1,69 @@
+namespace FsAutoComplete
+open System
+
+module FsProjEditor =
+
+  let moveFileUp (fsprojPath: string) (file: string) =
+    let xdoc = System.Xml.XmlDocument()
+    xdoc.Load fsprojPath
+    let xpath = sprintf "//Compile[@Include='%s']/.." file
+    let itemGroup = xdoc.SelectSingleNode(xpath)
+    let childXPath = sprintf "//Compile[@Include='%s']" file
+    let node = itemGroup.SelectSingleNode(childXPath)
+    let upNode = node.PreviousSibling
+    if isNull upNode then ()
+    else
+      itemGroup.RemoveChild node |> ignore
+      itemGroup.InsertBefore(node, upNode) |> ignore
+      xdoc.Save fsprojPath
+
+  let moveFileDown (fsprojPath: string) (file: string) =
+    let xdoc = System.Xml.XmlDocument()
+    xdoc.Load fsprojPath
+    let xpath = sprintf "//Compile[@Include='%s']/.." file
+    let itemGroup = xdoc.SelectSingleNode(xpath)
+    let childXPath = sprintf "//Compile[@Include='%s']" file
+    let node = itemGroup.SelectSingleNode(childXPath)
+    let downNode = node.NextSibling
+    if isNull downNode then ()
+    else
+      itemGroup.RemoveChild node |> ignore
+      itemGroup.InsertAfter(node, downNode) |> ignore
+      xdoc.Save fsprojPath
+
+  let addFileAbove (fsprojPath: string) (aboveFile: string) (newFileName: string) =
+    let xdoc = System.Xml.XmlDocument()
+    xdoc.Load fsprojPath
+    let xpath = sprintf "//Compile[@Include='%s']/.." aboveFile
+    let itemGroup = xdoc.SelectSingleNode(xpath)
+    let childXPath = sprintf "//Compile[@Include='%s']" aboveFile
+    let aboveNode = itemGroup.SelectSingleNode(childXPath)
+    let node = aboveNode.Clone ()
+    let attr = node.Attributes.Item 0
```

Comment:
is this overwriting the `Include` attribute on the new node? is it possible to index by attribute name instead to prevent ordering issues?